Welcome screen

sdwowk opened the pull request.

Commit:
Added welcome screen layout

Using the storyboard I created the welcome screen layout.

Commit:
Create fragment class for username

Commit:
Started work on the username fragment

Commit:
Added functionality (onclick listeners) to files

Commit:
Adding Stefan's things to my branch

Commit:
Merged my string.xml file with Stefan's branch

linster commented:
I approve this message.

stephenjust commented on QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/MainActivity.java:
```diff
@@ -61,4 +68,15 @@ public boolean onOptionsItemSelected(MenuItem item) {
+    		String username = intent.getStringExtra("username");
+    		Toast.makeText(this, "Welcome " + username + " to Qasper", Toast.LENGTH_SHORT).show();
+    		
+    		DataManager.getLocalDataStore().setUsername(username);
```
Instead of making the datastore handle only local data, I feel like we could be using an android Settings list type thing for only-local values. That way it keeps the IDataStore interfaces the same for local and remote. Thoughts?

sdwowk replied:
I can throw this out I was just using the current model of the datamanger that we have so far.

stephenjust replied:
Yeah I know. I'm just thinking ahead a bit. It would probably be easier to
synchronize between local and remote data if they were exactly the same,
and store local-specific data some other way.
On Oct 23, 2014 4:49 PM, "sdwowk" notifications@github.com wrote:

> In
> QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/MainActivity.java:
> 
> > @@ -61,4 +68,15 @@ public boolean onOptionsItemSelected(MenuItem item) {
> >          }
> >          return super.onOptionsItemSelected(item);
> >      }
> > +
> > -    public void onActivityResult(int requestCode, int resultCode,Intent intent){
> > -   
> > -       if (requestCode == Activity.RESULT_FIRST_USER || resultCode == Activity.RESULT_OK){
> > -           String username = intent.getStringExtra("username");
> > -           Toast.makeText(this, "Welcome " + username + " to Qasper", Toast.LENGTH_SHORT).show();
> > -   
> > -           DataManager.getLocalDataStore().setUsername(username);
> 
> I can throw this out I was just using the current model of the datamanger
> that we have so far.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/CMPUT301F14T14/Project/pull/56/files#r19313623.

Commit:
Have dialog fragment for username opening

Stefan and I have got the dialog fragment working but not returning to
Main Activity or returning the username

Commit:
Welcome screen returns username to MainActivity

Commit:
Add username to the data manager

stephenjust commented on QuestionApp/src/ca/ualberta/cs/cmput301f14t14/questionapp/DataManager.java:
```diff
@@ -8,7 +8,7 @@
 
 public class DataManager {
-	private LocalDataStore localDataStore;
+	private static LocalDataStore localDataStore;
```
Don't do static. Make this class a singleton instead.

stephenjust replied:
Also this is one of the places that it makes sense to declare both data stores as IDataStore. Except that won't work while LocalDataStore has some specific state associated with it.

stephenjust commented:
I think I'm done feedback-ing. If you need clarification, let me know. If you don't agree with anything, let me know.

sdwowk commented:
Sounds good, implementation for DataManager isn't in planning till next week. I got rid of the welcome screen's dependence on it so that we can merge it to the master branch.

Commit:
Got rid of Datamanager bit, added title to fragment
